### DISCUSSION TITLE
Password authentication stopped working on Arch Linux ARM package

### DISCUSSION THREAD

**Discussion by eh8** [transmission/transmission#2631] [Support]
I'm using the Arch Linux ARM [Transmission package](https://archlinuxarm.org/packages/armv7h/transmission-cli).

Ever since the recent update to 3.00-3, all password authentication attempts password are failing. Issue occurs on both the web UI as well as the shell (using the command `transmission-remote ...`)

I have:

- ensured that the Transmission daemon is stopped before modifying `settings.conf` to reset passwords
- reproduced the issue on another Arch system
- used passwords of various lengths and character combinations (i.e. letters only, no special characters, etc).
- verified that my `settings.conf` was intact and had no duplicated key-value pairs, broken syntax, or other strange irregularities
- built `transmission-cli` from source
- uninstalled `transmission-cli`, purged all configuration files, and reinstalled 
`transmission-cli`

Best workaround for now is to disable rpc authentication, but I don't want clients on my network to have partial read/write access to my system via the web UI.

**Comment by ckerr** [comment]
Yuck.

My first guess is either configuration or packaging error, since 3.00 hasn't been touched in a long time; there have been no official patch / point releases since then.

I don't see anything in the downstream [3.00-3 packaging diff](https://archlinuxarm.org/packages/armv7h/transmission-cli/log?id=7883915ed37f603abf5bbb7acf86b7537b4063eb) that screams out at me as a red flag.

Just to be clear, did it work for you before the 3.00-3 upgrade? From your phrasing ("Ever since the recent update") it sounds like it. If you downgrade to 3.00-2 does it work?

**Comment by eh8** [comment]
Yes, I didn't have any issues before. I also cloned the Arch Linux repo and built 3.00-2 as a sanity check and found that authentication is working as expected. Upgrading to 3.00-3 subsequently breaks authentication.

My hunch is the newly added flags that signal the compiler to use distribution libraries are inadvertently causing 32-bit systems to misbehave in operation.

I built the package from source again under different compilation instructions to see if this was true:

1. **Disabling fat link-time optimization (~~`export CFLAGS+=" -ffat-lto-objects"`~~)**

Still broken.

2. **Stop using external `libb64` (~~`--enable-external-b64`~~)**

Authentication works normally! 🎉

3. **Disabling fat link-time optimization _and_ stop using external `libb64`**

Authentication works normally, but probably because of the `libb64` tweak.

---

If you're a similar Arch Linux ARM administrator with machines on the armv7h architecture, then be sure to edit the PKGBUILD and remove the `--enable-external-b64` line when updating.

**Comment by eh8** [comment]
I found this [Arch Linux ARM forum thread](https://archlinuxarm.org/forum/viewtopic.php?f=57&t=15868&hilit=transmission) after I confirmed the workaround, some members found the specific `libb64` oddity that was causing authentication to fail.

So there's now two workarounds:

1. Install `libb64-git` from the AUR, which is a patched version that will enable Transmission to function properly on Arch Linux ARM. Then (re)install `transmission-cli` from the official repositories and restart the daemon.
2. If you prefer to rebuild the package yourself, then modify the PKGBUILD and delete the `--enable-external-b64` line.

**Reply by gustavopicheki** [reply]
Thank you so much! It fixed my problem!
